static/js/main.js

```diff
@@ -127,6 +128,45 @@ function goBack(){
     table.innerHTML = tabletext
 }
 
+
+function getNodes(url){
+    var jsonobj = ''
+    var xhttp = ''
+    try{
+        xhttp = new XMLHttpRequest()
+    } catch(e){
+        alert('It looks like you are using an unsupported browser. \nAOL tech support number 1-800-827-6364')
+    }
+    xhttp.onreadystatechange = function(){
+        if(xhttp.readyState == 4 && xhttp.status == 200){
```

Comment:
If you are going to use `onreadystatechange` instead of `onload`, then please change this `4` to `XMLHttpRequest.DONE`. It is a lot more clear what your intentions are that way.

Reply:
Hey what book or tutorial or lectures did you learn this stuff with?  Cause every one I've seen has used those codes.  Thinking maybe I've been using old documentation/training.

Reply:
It is common practice to avoid using magic numbers when you can. I had to look up what state 4 is because there is no meaning behind just the number 4, and when I did that I easily found the associated constant name. If you change the 4 to the defined constant, `XMLHttpRequest.DONE`, then it becomes extremely obvious what you are checking for.

As for the onload thing, most of the JS I've seen and written uses that. I do not think it is more supported than `onreadystatechange` though, since that method has been around since the start. If we don't care about super far back compatibility then I would prefer using onload for simplicity, but this works and I don't care too much.